### ISSUE TITLE
TriviaQA Source/Target Files

### ISSUE THREAD

**Issue by ronakice** [morningmoni/GAR#15]
Hey, is it possible to provide your TriviaQA source/target files at least for the train set? Or the process to generate them.

**Comment by morningmoni** [comment]
The [data](https://drive.google.com/file/d/15JYV5nnZNSMJNVV7OwZioc4NIljvWgHq/view?usp=sharing) for GAR training/testing on TriviaQA is uploaded.

**Comment by ronakice** [comment]
Thanks @morningmoni ! Another question, how do you generate the train.source/train.target files exactly?

**Comment by ronakice** [comment]
This is for NQ and TQA both

**Comment by morningmoni** [comment]
The train.source is always the question. 

For train.target --

For answer, they are provided by the dataset (NQ as it is; we take the value field for TQA and normalize it to first-letter-capital word if it was all-capital: ABC -> Abc)
  
For title and sentence, we use the original question as query and retrieve via BM25, then take the (title of / sentence in) the first passage that contains an answer. You may also try the gold passage is available.

**Comment by ronakice** [comment]
Thanks @morningmoni  One more question, I was unable to replicate properly the GAR BM25 results for both NQ and TriviaQA (the drift was more for TriviaQA with results just about standard pyserini BM25). Did you only expand those queries for which the source files are provided (that is some weren't expanded with titles or sentences)? Because in my case I expand all but the results fail to match. Also, was there any post-processing after generation or just direct appending?

**Comment by morningmoni** [comment]
No, we expanded all the queries actually. And it was direct appending. 

Maybe you can first try skip training and do BM25 using the provided query files to see if you can get similar results. Below are the results reproduced by someone else using pyserini BM25 on NQ:

top-k | answer |   | sentence |   | title |   | combined
-- | -- | -- | -- | -- | -- | -- | --
1 | 36.8% |   | 36.7% |   | 29.5% |   | 36.8%
5 | 53.3% |   | 53.0% |   | 51.6% |   | 60.8%
10 | 61.2% |   | 60.9% |   | 59.7% |   | 68.1%
20 | 67.6% |   | 67.2% |   | 67.9% |   | 74.8%
50 | 75.3% |   | 75.0% |   | 76.3% |   | 81.6%
100 | 80.0% |   | 80.1% |   | 80.9% |   | 85.2%

For TriviaQA, GAR-answer could be similar to standard BM25 individually except for top-1 (48.4 vs 53.2 in our experiments) as we found BM25 particularly strong on TriviaQA. However, when you combine the results of all the 3 sources, you should get better performance.

**Comment by shuishen112** [comment]
I am not sure how you combined the result.  Did you use the  [trectools](https://github.com/joaopalotti/trectools) to fuse the final three runs?

One more question, In your paper, " we fuse the retrieval results in astraightforward way: an equal number of passages are taken from the top-retrieved passages of eachsource". What is your meaning of "an equal number"?

**Comment by morningmoni** [comment]
> I am not sure how you combined the result. Did you use the [trectools](https://github.com/joaopalotti/trectools) to fuse the final three runs?
> 
> One more question, In your paper, " we fuse the retrieval results in astraightforward way: an equal number of passages are taken from the top-retrieved passages of eachsource". What is your meaning of "an equal number"?

No, we didn't use the trectools in the paper (but it might lead to slightly better results).

Equal number means interleaving - you take the top-1 retrieved document from each source, then top-2, till say top-33 (100 documents in total).

**Comment by shuishen112** [comment]
Thanks for explaination. But I am also a little confused. Could you please share your script of "combined"? Thank you very much.

**Comment by shuishen112** [comment]
Also, I used the provided query files to see results. I also used pyserini BM25 on NQ:

For answer:
Top1    accuracy: 0.35678670360110804
Top5    accuracy: 0.5321329639889196
Top10   accuracy: 0.6083102493074792
Top20   accuracy: 0.6709141274238227
Top50   accuracy: 0.7506925207756233
Top100  accuracy: 0.7944598337950138

For sentence:

Top1    accuracy: 0.36121883656509696
Top5    accuracy: 0.5274238227146815
Top10   accuracy: 0.6041551246537397
Top20   accuracy: 0.6662049861495845
Top50   accuracy: 0.7470914127423822
Top100  accuracy: 0.7952908587257618

For title:

Top1    accuracy: 0.292797783933518
Top5    accuracy: 0.5135734072022161
Top10   accuracy: 0.5950138504155125
Top20   accuracy: 0.6720221606648199
Top50   accuracy: 0.7559556786703601
Top100  accuracy: 0.8024930747922437

The result is a little different. is it normal?

**Comment by morningmoni** [comment]
> Thanks for explaination. But I am also a little confused. Could you please share your script of "combined"? Thank you very much.

Let me explain in more detail. Say you have 3 lists of retrieved docs [a1, a2, ...], [b1, b2, ...], [c1, c2, ...].
The combined list would be [a1, b1, c1, a2, b2, c2, ...]. Is it clearer?

**Comment by morningmoni** [comment]
> Also, I used the provided query files to see results. I also used pyserini BM25 on NQ:
> 
> For answer: Top1 accuracy: 0.35678670360110804 Top5 accuracy: 0.5321329639889196 Top10 accuracy: 0.6083102493074792 Top20 accuracy: 0.6709141274238227 Top50 accuracy: 0.7506925207756233 Top100 accuracy: 0.7944598337950138
> 
> For sentence:
> 
> Top1 accuracy: 0.36121883656509696 Top5 accuracy: 0.5274238227146815 Top10 accuracy: 0.6041551246537397 Top20 accuracy: 0.6662049861495845 Top50 accuracy: 0.7470914127423822 Top100 accuracy: 0.7952908587257618
> 
> For title:
> 
> Top1 accuracy: 0.292797783933518 Top5 accuracy: 0.5135734072022161 Top10 accuracy: 0.5950138504155125 Top20 accuracy: 0.6720221606648199 Top50 accuracy: 0.7559556786703601 Top100 accuracy: 0.8024930747922437
> 
> The result is a little different. is it normal?

It looks slightly worse but I think the range is reasonable.

**Comment by shuishen112** [comment]
> > Thanks for explaination. But I am also a little confused. Could you please share your script of "combined"? Thank you very much.
> 
> Let me explain in more detail. Say you have 3 lists of retrieved docs [a1, a2, ...], [b1, b2, ...], [c1, c2, ...]. The combined list would be [a1, b1, c1, a2, b2, c2, ...]. Is it clearer?

Thanks. This is my fusion result.

Top5    accuracy: 0.6044321329639889
Top10   accuracy: 0.678393351800554
Top20   accuracy: 0.7465373961218836
Top50   accuracy: 0.813573407202216
Top100  accuracy: 0.8457063711911358

**Comment by shuishen112** [comment]
One more quesiton, In your paper, you use the same method to fuse the GAR and DPR. One list is from GAR(My fusion result from three source)[a1,a2,a3], One list from DPR[b1,b2,b3]. Then we also combied the list [a1,b1,a2,b2,a3,b3].  Am I right? I have tried to fuse the result of GAR and DPR. But the result seems not as well as your result in the paper, This is my fusion result of DPR and GAR: 

Top5    accuracy: 0.6706371191135734
Top10   accuracy: 0.7429362880886426
Top20   accuracy: 0.7947368421052632
Top50   accuracy: 0.8379501385041551
Top100  accuracy: 0.8609418282548477
